modular_skyrat/modules/modular_weapons/code/company_and_or_faction_based/carwo_defense_systems/rifle.dm

```diff
@@ -0,0 +1,156 @@
+		use by SolFed's various infantry branches. Following the rather reasonable \
+		military requirements of using the same few cartridges and magazines, \
+		the lifespans of logistics coordinators and quartermasters everywhere \
+		were lengthened by several years. While typically only for military sale \
+		in the past, the recent collapse of certain unnamed weapons manufacturers \
+		has caused Carwo to open many of its military weapons to civilian sale, \
+		which includes this one."
+
+	return .
+
+/obj/item/gun/ballistic/automatic/sol_rifle/no_mag
+	spawnwithmagazine = FALSE
+
+// Sol marksman rifle
+
+/obj/item/gun/ballistic/automatic/sol_rifle/marksman
+	name = "\improper Carwo 'd'Elite' Marksman Rifle"
+	desc = "A heavy marksman rifle commonly seen in the hands of SolFed military types. Accepts any standard SolFed rifle magazine."
+
+	icon_state = "elite"
+	worn_icon_state = "elite"
+	inhand_icon_state = "elite"
+
+	spawn_magazine_type = /obj/item/ammo_box/magazine/c40sol_rifle
+
+	fire_delay = 3
+
+	spread = 0
+
+/obj/item/gun/ballistic/automatic/sol_rifle/marksman/Initialize(mapload)
+	. = ..()
+
+	AddComponent(/datum/component/scope, range_modifier = 2)
+
+/obj/item/gun/ballistic/automatic/sol_rifle/marksman/give_autofire()
+	return
+
+/obj/item/gun/ballistic/automatic/sol_rifle/marksman/examine_more(mob/user)
+	. = ..()
+
+	. += "This particlar variant, often called 'd'Elite', is a marksman rifle. \
+		Automatic fire was forsaken for a semi-automatic setup, a more fitting \
+		stock, and more often than not a scope. Typically also seen with smaller \
+		magazines for convenience for the shooter, but as with any other Sol \
+		rifle, all standard magazine types will work."
+
+	return .
+
+/obj/item/gun/ballistic/automatic/sol_rifle/marksman/no_mag
+	spawnwithmagazine = FALSE
+
+// Machinegun based on the base Sol rifle
+
+/obj/item/gun/ballistic/automatic/sol_rifle/machinegun
+	name = "\improper Carwo 'd'Outomaties' Machinegun"
+	desc = "A hefty machinegun commonly seen in the hands of SolFed military types. Accepts any standard SolFed rifle magazine."
+
+	icon_state = "outomaties"
+	worn_icon_state = "outomaties"
+	inhand_icon_state = "outomaties"
+
+	bolt_type = BOLT_TYPE_OPEN
+
+	spawn_magazine_type = /obj/item/ammo_box/magazine/c40sol_rifle/drum
+
+	fire_delay = 1
+
+	recoil = 1
+	spread = 10
+
+/obj/item/gun/ballistic/automatic/sol_rifle/machinegun/examine_more(mob/user)
+	. = ..()
+
+	. += "The d'Outomaties variant of the rifle, what you are looking at now, \
+		is a modification to turn the weapon into a passable, if sub-optimal \
+		light machinegun. To support the machinegun role, the internals were \
+		converted to make the gun into an open bolt, faster firing machine. These \
+		additions, combined with a battle rifle not meant to be used fully auto \
+		much to begin with, made for a relatively unwieldy weapon. A machinegun, \
+		however, is still a machinegun, no matter how hard it is to keep on target."
```

Comment:
This type of string separation is so much easier to read through. Not that I have, because I have so many files to read through, but y'know. Thank you.